developer-workflow/extension-modules.rst

```diff
@@ -5,13 +5,336 @@
 Standard library extension modules
 ==================================
```

Comment:
Titles should follow this guidelines: https://devguide.python.org/documentation/markup/#sections

```suggestion
==================================
Standard library extension modules
==================================
```

In addition, depending on the differences between "CPython extensions" and "extensions module" (see my other comment), you might want to have a section for each, with the one for "CPython extensions" only having a short paragraph and a list of links.  If there is no meaningful difference, one of the top-levels of the page could be removed.  For reference, this is the current structure of the page:
```
Standard library extension modules
    Adding an extension module to CPython
        Make the CPython project compile
        Compile the CPython project
        Troubleshooting
            make regen-configure does not work!
```

Reply:
I should have read the style guide beforehand... 

>  If there is no meaningful difference, one of the top-levels of the page could be removed.

For me, there is none. Unless you want to make the distinction between "CPython extension" for the extensions part of the CPython project only.

Ideally, I'd like to have the following sections:

```text
Standard library extension modules
    Adding an extension module to CPython
        Make the CPython project compile
        Compile the CPython project
        Troubleshooting
            make regen-configure does not work!

    Adding a frozen module to CPython
        Make the CPython project compile
        Compile the CPython project
        Troubleshooting

	Adding a configuration value to CPython (like Py_HAVE_C_COMPLEX)
``` 

The assumptions on frozen modules are distinct and I also wanted to add something about how to add configuration values. For me "Standard library extension modules" referred to the CPython project itself and how to extend the C implementation itself (whatever is in Modules/ (maybe Objects/ and Include/) but not in Lib/).

The section "Adding to the stdlib" only focuses on the process but not how to make the project work after (like where to propose things, etc but not the technical aspects). There is also a separte page for the Python C API but I think it's also something separate (I mean, you cannot change the C API with new files if you don't know how to...).

Reply:
Btw, the title already has '===' above. It's just that GitHub didn't show it in the diff =/

Reply:
> Btw, the title already has '===' above. It's just that GitHub didn't show it in the diff =/

My bad, I didn't notice there were a few more lines above.  I removed the suggestion now.

Also no worries about the style guide -- that's what reviewers are for :)

Regarding the distinction between "CPython extensions" and "extension modules", we should hear from someone that is more familiar with those (@ncoghlan, @matrixise?).  The rest of your plan sounds good to me.

Reply:
Actually, by digging into the issue I mentioned (missing the configure/bootstrap) I am now confused about the terminology. I *think* the accelerator modules are actually referred to as *built-in modules* and some other modules such as blake2 are *extension* modules.

Also, it's still unclear to me which files must be manually edited and which ones need to be autogenerated. AFAIK, I managed to make my PR https://github.com/python/cpython/pull/121446 work but I'm pretty sure that this is perhaps not the way to go :') So I would be very happy if someone could tell me the *exact* steps to do (the solution I propose should work but it's probably not the best one?)

Reply:
Update: frozen modules are actually written in Python, so they wouldn't be part of this tutorial although incoporating them into CPython is done at the C level.